packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts

```diff
@@ -203,53 +190,58 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
     try {
       const indyVdrPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService)
 
-      const did = didFromCredentialDefinitionId(credentialDefinitionId)
-
-      const pool = await indyVdrPoolService.getPoolForDid(agentContext, did)
+      // we support did:indy and legacy identifiers
+      const { did, didIdentifier, schemaSeqNo, tag } = parseCredentialDefinitionId(credentialDefinitionId)
+      const { pool } = await indyVdrPoolService.getPoolForDid(agentContext, did)
 
       agentContext.config.logger.debug(
         `Getting credential definition '${credentialDefinitionId}' from ledger '${pool.indyNamespace}'`
       )
 
+      const legacyCredentialDefinitionId = getLegacyCredentialDefinitionId(didIdentifier, schemaSeqNo, tag)
       const request = new GetCredentialDefinitionRequest({
-        submitterDid: did,
-        credentialDefinitionId,
+        credentialDefinitionId: legacyCredentialDefinitionId,
       })
 
       agentContext.config.logger.trace(
         `Submitting get credential definition request for credential definition '${credentialDefinitionId}' to ledger '${pool.indyNamespace}'`
       )
-
       const response = await pool.submitReadRequest(request)
 
-      const schema = await this.fetchIndySchemaWithSeqNo(agentContext, response.result.ref, did)
+      // We need to fetch the schema to determine the schemaId (we only have the seqNo)
+      const schema = await this.fetchIndySchemaWithSeqNo(agentContext, response.result.ref, didIdentifier)
```

Comment:
Would it make sense for AnonCredsRegistrars to first check in the AnonCredsSchemaRepository so no extra transaction is required? And also store it after retrieving? Or this is something that should be done in AnonCredsApi level? 

This is not strictly related to this PR but was something that felt a bit awkward when doing integration with AATH.

Reply:
We could check in the schema repository. I'm not sure about storing it as the records are for things this agent created, not for just any schema record.

I think not having to fetch the schema is something we could solve using the cache.

Reply:
We can't actually look at the schemaRecord as we don't store the seqNo. We should probably add it to the metadata, but I wanted to have as little indy specific logic as possible. 

I'll address this in a follow up PR

Reply:
Good point! Of course not needed for this PR, just some thinking that came up to my head while reviewing this one.